sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,374 @@
+ *     - `<namespace-component>/<topic-name>/Subscriptions/<subscription-name>`
+ *     - `<namespace-component>/<any-entity-name>`
+ *
+ * @param entry
+ * @param nameProperties
+ */
+function setName(entry: any, nameProperties: any): any {
+  if (entry[Constants.ATOM_METADATA_MARKER]) {
+    let rawUrl = entry[Constants.ATOM_METADATA_MARKER].id;
+
+    // The parsedUrl gets constructed differently for browser vs Node.
+    // It is specifically behaves different for some of the Atom based management API where
+    // the received URL in "id" element is of type "sb:// ... " and not a standard HTTP one
+    // Hence, normalizing the URL for parsing to work as expected in browser
+    if (rawUrl.startsWith("sb://")) {
+      rawUrl = "https://" + rawUrl.substring(5);
+    }
+
+    const parsedUrl = new URL(rawUrl);
+    const pathname: string = parsedUrl.pathname;
+
+    const firstIndexOfDelimiter = pathname.indexOf("/");
+
+    if (pathname.match("(.*)/(.*)/Subscriptions/(.*)/Rules/(.*)")) {
+      const lastIndexOfSubscriptionsDelimiter = pathname.lastIndexOf("/Subscriptions/");
+      const firstIndexOfRulesDelimiter = pathname.indexOf("/Rules/");
+      entry[nameProperties[0]] = pathname.substring(
+        firstIndexOfDelimiter + 1,
+        lastIndexOfSubscriptionsDelimiter
+      );
+      entry[nameProperties[1]] = pathname.substring(
+        lastIndexOfSubscriptionsDelimiter + 15,
+        firstIndexOfRulesDelimiter
+      );
+      entry[nameProperties[2]] = pathname.substring(firstIndexOfRulesDelimiter + 7);
+    } else if (pathname.match("(.*)/(.*)/Subscriptions/(.*)")) {
+      const lastIndexOfSubscriptionsDelimiter = pathname.lastIndexOf("/Subscriptions/");
+      entry[nameProperties[0]] = pathname.substring(
+        firstIndexOfDelimiter + 1,
+        lastIndexOfSubscriptionsDelimiter
+      );
+      entry[nameProperties[1]] = pathname.substring(lastIndexOfSubscriptionsDelimiter + 15);
+    } else if (pathname.match("(.*)/(.*)")) {
+      entry[nameProperties[0]] = pathname.substring(firstIndexOfDelimiter + 1);
+    }
+  }
+}
+
+/**
+ *
+ * Utility to help construct the normalized `RestError` object based on given `errorBody`
+ * data and other data present in the received `response` object.
+ *
+ * @param errorBody
+ * @param response
+ */
+export function buildError(errorBody: any, response: HttpOperationResponse): RestError {
+  const normalizedError: any = {};
+  const odataErrorFormat = !!errorBody["odata.error"];
+  const errorProperties =
+    errorBody.Error || errorBody.error || errorBody["odata.error"] || errorBody;
+  let errorMessage;
+
+  if (typeof errorBody === "string") {
+    errorMessage = errorBody;
+  } else {
+    if (odataErrorFormat) {
+      Object.keys(errorProperties).forEach((property: string) => {
+        let value = errorProperties[property];
+        if (property === Constants.ODATA_ERROR_MESSAGE && typeof value !== "string") {
+          if (value[Constants.ODATA_ERROR_MESSAGE_VALUE]) {
```

Comment:
A few lines below for the case where we are not dealing with odata error format, we check if `value` exists before trying to access its members. Same check here is useful.

```suggestion
          if (value && value[Constants.ODATA_ERROR_MESSAGE_VALUE]) {
```